test/optim/test_optimize.py

```diff
@@ -1520,10 +1539,24 @@ def test_optimize_acqf_mixed_q2(self, mock_optimize_acqf):
             candidate_rvs, exp_candidates, acq_val_rvs = [], [], []
             # generate mock side effects and compute expected outputs
             for _ in range(q):
-                candidate_rvs_q = [torch.rand(1, 3, **tkwargs) for _ in range(num_ff)]
-                acq_val_rvs_q = [torch.rand(1, **tkwargs) for _ in range(num_ff)]
-                best = torch.argmax(torch.stack(acq_val_rvs_q))
-                exp_candidates.append(candidate_rvs_q[best])
+                candidate_rvs_q = [
+                    torch.rand(num_restarts, 1, 3, **tkwargs) for _ in range(num_ff)
+                ]
+                acq_val_rvs_q = [
+                    torch.rand(num_restarts, **tkwargs) for _ in range(num_ff)
+                ]
+
+                best_acq_values = torch.tensor(
+                    [torch.max(acq_values) for acq_values in acq_val_rvs_q]
+                ).to(acq_val_rvs_q[0])
```

Comment:
Don't think this `to` is needed here
```suggestion
                )
```